pkg/controlplane/apiserver/config.go

```diff
@@ -180,6 +180,10 @@ func BuildGenericConfig(
 		sets.NewString("attach", "exec", "proxy", "log", "portforward"),
 	)
 
+	if genericConfig.RequestTimeout > 0 {
```

Comment:
@sttts - would you mind taking a look at the second commit - in particular this line and suggest if there is a better place for achieving this?

Reply:
There is prior work using this pattern below. So adding another one here is not the end of the world. 

But in general, mutating `s` here in the config constructor has smell (unexpected side-effect). 

The cleaner solution, and question is whether there are other ripples, would be to move this kind of cross-options logic into the option completion func, i.e. here for controlplane/apiserver that would be https://github.com/kubernetes/kubernetes/blob/95d71c464a6a5f81ae36bddc86972b3b2a1c40a5/pkg/controlplane/apiserver/options/options.go#L206.

Reply:
The pattern is always:

NewOptions, tweak values, Complete().
NewConfig(options), tweak values, Complete().
NewServer(config), tweak values, Complete(), PrepareRun(), Run().

The `Complete()` func is where you add defaulting, or cross options logic. Every constructor is side-effect-free.